test/functional/ui/screen.lua

```diff
@@ -202,9 +202,11 @@ function Screen:expect(expected, attr_ids, attr_ignore)
     for i = 1, self._height do
       local expected_row = expected_rows[i]
       local actual_row = self:_row_repr(self._rows[i], ids, ignore)
-      if expected_row ~= actual_row then
-        return 'Row '..tostring(i)..' didnt match.\nExpected: "'..
-               expected_row..'"\nActual:   "'..actual_row..'"'
+      if not expected_row then
+        return 'Screen has more rows than expected.'
```

Comment:
During test development, if a `screen:expect()` call has too few lines, `expected_row` is nil. So this shows a better error message in that case.

Perhaps we should also do a sanity check for too *many* expected rows.